ddl/reorg.go

```diff
@@ -613,18 +619,26 @@ func getReorgInfo(ctx *JobContext, d *ddlCtx, rh *reorgHandler, job *model.Job,
 		} else {
 			tb = tbl.(table.PhysicalTable)
 		}
-		start, end, err = getTableRange(ctx, d, tb, ver.Ver, job.Priority)
-		if err != nil {
-			return nil, errors.Trace(err)
+		if mergingTmpIdx {
+			start, end = tablecodec.GetTableIndexKeyRange(pid, tablecodec.TempIndexPrefix|elements[0].ID)
+		} else {
+			start, end, err = getTableRange(ctx, d, tb, ver.Ver, job.Priority)
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
 		}
 		logutil.BgLogger().Info("[ddl] job get table range",
 			zap.Int64("jobID", job.ID), zap.Int64("physicalTableID", pid),
-			zap.String("startHandle", tryDecodeToHandleString(start)),
-			zap.String("endHandle", tryDecodeToHandleString(end)))
+			zap.String("startKey", hex.EncodeToString(start)),
+			zap.String("endKey", hex.EncodeToString(end)))
 
 		failpoint.Inject("errorUpdateReorgHandle", func() (*reorgInfo, error) {
 			return &info, errors.New("occur an error when update reorg handle")
 		})
+		err = rh.RemoveDDLReorgHandle(job, elements)
```

Comment:
Because the reorg handle is reused in the merge process. If we don't remove it, `InitDDLReorgHandle` can report the duplicate entry error.

Reply:
May we can put it to https://github.com/pingcap/tidb/pull/37709/files#diff-51926b25b437247805c6880b2fd041f096575b30050b560ab934f1d863bb6bf6R622, I think it'll make it easier to understand.

Reply:
Another reason need to remove this handler is current we do not have  checkpoint, if lightning way failed, we have to restart the reorg task. we will remove it when checkpoint supported.

Reply:
There is another case need this: when the lightning backfill worker is WIP and the TiDB server halts, the whole process need to be restarted(because the checkpoint is not implemented yet). If we don't remove the reorg handle when we restart, it enters a dead loop.